aFWall/src/main/java/dev/ukanth/ufirewall/util/LogNetUtil.java

```diff
@@ -0,0 +1,307 @@
+package dev.ukanth.ufirewall.util;
+
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+
+import com.afollestad.materialdialogs.DialogAction;
+import com.afollestad.materialdialogs.MaterialDialog;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+
+import dev.ukanth.ufirewall.Api;
+import dev.ukanth.ufirewall.R;
+import dev.ukanth.ufirewall.log.Log;
+import eu.chainfire.libsuperuser.Shell;
+
+/**
+ * This file was created to simplify Network Function in AfWall+ log system
+ * Created by vzool on 1/20/17.
+ */
+
+public class LogNetUtil {
+
+    final static String TAG = "afWall-LogNetUtil";
+
+    public static class NetTask extends AsyncTask<NetParam, Integer, String>
+    {
+        long start_time;
+        OnFinishRequest onFinishRequest_callback;
+        MaterialDialog progress;
+        Context context;
+        String output_result = "";
+
+        private static final String PING_CMD = "%s ping -w 1 -W %d %s";
+
+        public NetTask(Context context){
+            this.context = context;
+        }
+
+        long finish_time(){
+            return System.currentTimeMillis() - start_time;
+        }
+
+        public NetTask setOnFinishRequest(OnFinishRequest when){
+            onFinishRequest_callback = when;
+            return this;
+        }
+
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+
+            progress = new MaterialDialog.Builder(context)
+                    .title(R.string.searching)
+                    .content(R.string.looking_for_data)
+                    .progress(true, 0)
+                    .progressIndeterminateStyle(true)
+                    .show();
+        }
+
+        @Override
+        protected String doInBackground(NetParam... params)
+        {
+            start_time = System.currentTimeMillis();
+
+            try{
+
+                switch(params[0].type){
+
+                    case PING:
+
+                        // Ping
+
+                        try{
+
+                            String shell_result = "";
+                            String command = "";
+
+                            try{
+
+                                // This command needs permission to allow
+                                // AFWall+ itself to has access to network
+                                // to work probably
+                                command = String.format(PING_CMD, "", G.logPingTimeout(), params[0].address);
+
+                                Log.d(TAG, "Execute CMD: " + command);
+                                Process process = Runtime.getRuntime().exec(command);
```

Comment:
No, I don't. :)
I made it a choice if device support ping in Runtime will run normally and it will require AFWall+ network access permission.
If device not support ping in Runtime it will fall down to use BusyBox version.
I did what you ask [before](https://github.com/ukanth/afwall/pull/667/files/d5ba4741f8a78ca7bac1b14e8f1319309d606dd8..651bc3635f372496190a52172b404b321edb3185#diff-12ea6a97fac8b52ce252916160aed8a4R92), ping by libsuperuser in **normal shell** does always require a root permission **with network access** for `0:(root) Apps running as root` category.
Is this what you prefer?